EIPS/eip-4361.md

```diff
@@ -0,0 +1,313 @@
+
+Additional requirements added by authors:
+1. The user must be presented a human-understandable interface prior to signing, mostly free of machine-targeted artifacts such as JSON blobs, hex codes (aside from the Ethereum address), and baseXX-encoded strings.
+2. The application server must be able to implement fully usable support for its end without forcing a change in the wallets.
+3. There must be a simple and straightforward upgrade path for both applications and wallets already using Ethereum account-based signing for authentication.
+4. There must be facilities and guidelines for adequate mitigation of Man-in-the-Middle (MITM) attacks, replay attacks, and malicious signing requests.
+
+### Design Goals
+1. Human-Friendly
+2. Simple to Implement
+3. Secure
+4. Machine Readable
+5. Extensible
+
+### Technical Decisions
+- Why EIP-191 (Signed Data Standard) over [EIP-712](https://eips.ethereum.org/EIPS/eip-712) (Ethereum typed structured data hashing and signing)
+    - EIP-191 is already broadly supported across wallets UX, while EIP-712 support for friendly user display is pending. **(1, 2, 3, 4)**
+        - https://github.com/LedgerHQ/app-ethereum/issues/105
+        - https://github.com/trezor/trezor-firmware/pull/1568
+    - EIP-191 is simple to implement using a pre-set prefix prior to signing, while EIP-712 is more complex to implement requiring the further implementations of a bespoke Solidity-inspired type system, RLP-based encoding format, and custom keccak-based hashing scheme. **(2)**
+    - EIP-191 produces more human-readable messages, while EIP-712 creates signing outputs for machine consumption, with most wallets not displaying the payload to be signed in a manner friendly to humans. **(1)**![](https://i.imgur.com/yvhz2mD.png)
+
+    - EIP-712 has the advantage of on-chain representation and on-chain verifiability, such as for their use in metatransactions, but this feature is not relevant for the specification's scope. **(2)**
+- Why not use JWTs? Wallets don't support JWTs. The keccak hash function is [not assigned by IANA](https://www.iana.org/assignments/jose/jose.xhtml) for use as a JOSE algorithm. **(2, 3)**
+- Why not use YAML or YAML with exceptions? YAML is loose compared to ABNF, which can readily express character set limiting, required ordering, and strict whitespacing. **(2, 3)**
+
+### Discussions and Acknowledgements
```

Comment:
```suggestion
### Discussions and Acknowledgements
```
This type of information should be tracked in the `discussions-to` thread, not in the EIP itself.

Reply:
Addressed in https://github.com/ethereum/EIPs/pull/4361/commits/e59daacf95ecf9747b47921f7ce7b865eb139f75